src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.ts

```diff
@@ -39,6 +43,10 @@ export class VirtualKeyboardComponent implements OnChanges {
       .finally(() => this.updatePhysicalKeyboardLayout())
   }
 
+  private setFallbackKeyboardLayoutMap(): void {
+    this.keyboardLayoutMap = this.defaultKeyboardLayoutMapService.createDefaultKeyboardLayoutMap()
+  }
```

Comment:
If you used dependency injection for the `PhysicalKeyboardLayoutService` then you wouldn't need this. The `PhysicalKeyboardLayoutService` would just itself have an `KeyboardLayoutMapService` injected to it.
This way `VirtualKeyboardComponent` does not need to know about `DefaultKeyboardLayoutMapService`, it just need to call `KeyboardLayoutService.createKeyboardLayout()`.

Reply:
This part is removed and not using DI is fixed in #47.